src/transaction/locator_sr.c

```diff
@@ -5347,12 +5347,16 @@ locator_update_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oid, OID
     {
       HEAP_OPERATION_CONTEXT update_context;
 
-      or_class_tde_algorithm (recdes, &tde_algo);
-      if (tde_algo != TDE_ALGORITHM_NONE && !tde_Cipher.is_loaded)
+      if (!OID_IS_ROOTOID (oid))
```

Comment:
How about checking it with class_oid together?

if (OID_IS_ROOTOID (class_oid) && !OID_IS_ROOTOID (oid))

Reply:
`OID_IS_ROOTOID (class_oid)` is always true here (see 3 lines above). Do you think it is needed to make the condition clear?

Reply:
I mean it looks good if the OID_IS_ROOTOID (class_oid) (3 lines above) and the !OID_IS_ROOTOID (oid) check together.

Reply:
Other jobs have to be done even if `!OID_IS_ROOTOID (oid)` except for the TDE error handling.

Reply:
Oh! I read the code incorrectly because of misreading of the indentation.